odl/contrib/param_opt/param_opt.py

```diff
@@ -0,0 +1,92 @@
+# Copyright 2014-2017 The ODL contributors
+#
+# This file is part of ODL.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public License,
+# v. 2.0. If a copy of the MPL was not distributed with this file, You can
+# obtain one at https://mozilla.org/MPL/2.0/.
+
+"""Method(s) to find optimal reconstruction parameter(s) w.r.t. given FOM."""
+
+import numpy as np
+import scipy.optimize
+
+__all__ = ('optimal_parameters', )
+
+
+def optimal_parameters(reconstruction, fom, phantoms, data,
+                       initial_param=0):
+    r"""Find the optimal parameters for a reconstruction method.
+
+    Notes
+    -----
+    For a forward operator :math:`A : X \to Y`, a reconstruction operator
```

Comment:
This function doesn't even seem to care about forward operators at all. It's all inference.

Are we sure we want to write this using forward operators?

Reply:
We can rewrite. But in that case we should maybe rename most of the parameters as well? Their names are somehow also taken from what they represent in an inverse problem.

(`reconstruction` -> `operator`, `phantom` -> `phantom` or `ground_truth`, `data` -> ???)

Reply:
This became complicated, lets leave it :D